DM-456 new lwm2m device registration updates (#393)
SoftwareAG/c8y-docs: content/protocol-integration/lwm2m-bundle/register-device.md

If PSK security mode is selected, additional settings are required

![LWM2M single device registration form - mandatory with psk](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page1-psk.png)

The second step is configuration of the bootstrap settings. This configuration is needed if the device could connect with bootstrap. Refer to [Bootstrap settings](#lwm2m-device-registration-bootstrap-settings) section for more details about the fields.

![LWM2M single device registration form - bootstrap](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page2.png)

In the third step it is possible to do other optional configurations. Refer to [Advanced settings](#lwm2m-device-registration-advanced-settings) section for more details about the fields.

![LWM2M single device registration form - advanced 1](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-1.png)
![LWM2M single device registration form - advanced 2](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-2.png)
![LWM2M single device registration form - advanced 3](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-3.png)

> After filling all applicable information for the device, click on register. Registration result will be shown right away.

Reviewer: fixed

![LWM2M single device registration form - success](/images/device-protocols/lwm2m/lwm2m-device-reg-single-success.png)

<a name="lwm2m-bulk-device-registration"></a>
### Bulk device registration

If there are number of devices to be registered at the same time, it would be more convenient to register the LWM2M devices in {{< product-c8y-iot >}} by uploading a CSV file with registration data in the bulk registration dialog in **Devices** > **Registration** > **Register device** > **Bulk device registration** > **LWM2M** in the Device Management application. The dialog contains also CSV template links.

![LWM2M bulk device registration form](/images/device-protocols/lwm2m/lwm2m-device-reg-bulk.png)

When you upload the CSV file, dialog displays the message about where to find the result.

![LWM2M bulk device registration form](/images/device-protocols/lwm2m/lwm2m-device-reg-bulk-dialog-result-message.png)

> The result of the bulk device registration will be displayed under LWM2M connector device object created for tenant.

Author: I'll create a new page connector device and put the ref. link here.
We can later use that page to explain migration operation as well.

> The result of the bulk device registration will be displayed under LWM2M connector device object created for tenant.

Author: actually I'll explain the migration op. as well with this PR

> The result of the bulk device registration will be displayed under LWM2M connector device object created for tenant.

Reviewer 2: okay. We can create a new heading for all the operations

![LWM2M bulk device registration connector operation](/images/device-protocols/lwm2m/lwm2m-device-reg-bulk-connector-op-with-1-duplicated.png)

### Device registration settings

> <a name="lwm2m-device-registration-mandatoy-settings"></a>

Reviewer: fixed.

</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">Endpoint client ID / endpoint id</td>
<td style="text-align: left">String</td>
<td style="text-align:left">Indicates the LWM2M client's "endpoint ID" in order to allow the LwM2M bootstrap to provision the bootstrap information for the LWM2M client. The endpoint ID has be to be <b>unique</b> across all tenants and must have the same value as the ID.</td>
<td style="text-align: left">Yes</td>
</tr>
<tr>
<td style="text-align:left">Security mode / securityMode</td>
<td style="text-align: left">String</td>
<td style="text-align:left">Determines the type of connection used by the LWM2M device. "NO_SEC" is used for unsecure connections which means that there is no security. It is highly recommended to always protect the LWM2M protocol. However, there are scenarios in which the LWM2M protocol is deployed in environments where the lower layer security mechanisms are provided. "PSK" is used for secure connections. With "PSK", the client and server have a common secret symmetric cryptography. Currently {{< product-c8y-iot >}} supports only "NO_SEC" and "PSK".</td>
<td style="text-align: left">Yes</td>
</tr>
<tr>

> <td style="text-align: left">LWM2M PSK key / lwm2m psk_key</td>

Author: In csv it was lwm2m psk_key before so we need to be careful with that change or we might need to support both.
Same situation for lwm2m psk_id and publicKeyOrId

> <td style="text-align: left">LWM2M PSK key / lwm2m psk_key</td>

Author: As discussed in the team, this will be changed to LWM2M Pre-Shared Key in registration and bootstrap UI and csv field will remain the same.

layout: redirect
---

The data provided during registration must enable LWM2M communication and holds information for factory bootstrap and client-initiated bootstrap.
In the factory bootstrap mode, the LWM2M client has been configured with the necessary bootstrap information prior to the deployment of the device.
The client-initiated bootstrap mode requires a LWM2M bootstrap server account pre-loaded in the LWM2M client.

> {{< product-c8y-iot >}} supports registration for **unsecured** and **PSK-secured** LWM2M devices allowing connection with **NO_SEC** and **PSK** mode respectively.

Reviewer 3: ```suggestion
{{< product-c8y-iot >}} supports registration for **unsecured** and **PSK-secured** LWM2M devices allowing connection with NO_SEC and PSK mode respectively.
```

<a name="lwm2m-bulk-device-registration"></a>
### Bulk device registration

If there is a number of devices to be registered at the same time, it is more convenient to use bulk device registration.

To register the LWM2M devices in {{< product-c8y-iot >}}, navigate to **Devices** > **Registration** in the Device Management application, click **Register device** at the top right and select **Bulk device registration** > **LWM2M** from the dropdown.
Upload a CSV file with the registration data in the resulting bulk registration dialog.
The dialog also contains CSV template links.
Refer to [Required settings](#lwm2m-device-registration-required-settings), [Bootstrap settings](#lwm2m-device-registration-bootstrap-settings) and [Advanced settings](#lwm2m-device-registration-advanced-settings) below for details about the fields.

{{< c8y-admon-info >}}
The maximum size allowed for the CSV file is 10 MB.
{{< /c8y-admon-info >}}

When you upload the CSV file, the dialog will display a confirmation message that tells you where to find the result.

![LWM2M bulk device registration form](/images/device-protocols/lwm2m/lwm2m-device-reg-bulk-dialog-result-message.png)

> The the bulk device registration operation will be displayed in the [LWM2M connector device](#connector-device) object created for the tenant.

Reviewer 3: ```suggestion
The bulk device registration operation will be displayed in the [LWM2M connector device](#connector-device) object created for the tenant.
```

![LWM2M bulk device registration connector operation](/images/device-protocols/lwm2m/lwm2m-connector-device-bulk-device-reg-res.png)

### Device registration settings

<a name="lwm2m-device-registration-required-settings"></a>
#### Required settings

The fields below must be contained to be able to establish a connection:
<table>
<col style="width:25%">
<col style="width:10%">
<col style="width:50%">
<col style="width:15%">
<thead>
<tr>

> <th style="text-align:left">Field / CSV field name</th>

Reviewer 3: @Oezge-Akin @sourajitsaha17 Wouldn´t it be clearer to have two columns for these to settings instead of one? Or is there a specific reason to have these two values in one column?

Author: yeah it can be two column instead.

<td style="text-align: left">Mandatory for PSK</td>
</tr>
<tr>
<td style="text-align: left">Bootstrap PSK key / bootstrap psk_key</td>
<td style="text-align: left">String</td>
<td style="text-align: left">The hex-encoded key used by the device for bootstrap connections in PSK mode.</td>
<td style="text-align: left">Mandatory for PSK</td>
</tr>
</tbody>
</table>

<a name="lwm2m-device-registration-bootstrap-settings"></a>
#### Bootstrap settings

<a name="registration-of-no-sec-device"></a>
##### Registration of NO_SEC devices

Unsecured devices connect during bootstrap connection and server connection through unsecured ports:

* **5683**: unsecure bootstrap connection
* **5783**: unsecure direct server connection

<a name="registration-of-psk-secured-device"></a>
##### Registration of PSK-secured devices

PSK-secured devices connect during a bootstrap connection and a server connection using a pre-shared key through secured ports:

* **5684**: PSK bootstrap connection
* **5784**: PSK direct server connection

PSK keys must be provided during the device registration.

> The file must contain the fields defined in [Mandatory settings](#lwm2m-device-registration-mandatory-settings).

Reviewer 2: We missed this part. The Mandatory settings changed to 'Required settings'

Author: oh eagle eyes

The third step covers optional configurations.
Refer to [Advanced settings](#lwm2m-device-registration-advanced-settings) below for details about the fields.

![LWM2M single device registration form - advanced 1](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-1.png)
![LWM2M single device registration form - advanced 2](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-2.png)
![LWM2M single device registration form - advanced 3](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-3.png)

After filling all applicable information for the device, click **Register**.
The UI will display a confirmation message.

![LWM2M single device registration form - success](/images/device-protocols/lwm2m/lwm2m-device-reg-single-success.png)

<a name="lwm2m-bulk-device-registration"></a>
### Bulk device registration

If there is a number of devices to be registered at the same time, it is more convenient to use bulk device registration.
To register the LWM2M devices in {{< product-c8y-iot >}}, navigate to **Devices** > **Registration** in the Device Management application and click **Register device** > **Bulk device registration** > **LWM2M** at the top right.

> Upload a CSV file with the registration data in the resulting bulk registration dialog.

Reviewer 2: We should mention that the max size of the CSV is 10MB

Author: Added the info now.

layout: redirect
---

The data provided during registration must enable LWM2M communication and holds information for factory bootstrap and client-initiated bootstrap.
In the factory bootstrap mode, the LWM2M client has been configured with the necessary bootstrap information prior to the deployment of the device.

> The client-initiated bootstrap mode requires a LWM2M bootstrap-server account pre-loaded in the LWM2M client.

Reviewer 3: ```suggestion
The client-initiated bootstrap mode requires a LWM2M bootstrap server account pre-loaded in the LWM2M client.
```

{{< product-c8y-iot >}} supports registration for **unsecured** and **PSK-secured** LWM2M devices allowing connection with **NO_SEC** and **PSK** mode respectively.

You can register a LWM2M device in {{< product-c8y-iot >}} in two ways:

* [Single device registration](#lwm2m-single-device-registration)
* [Bulk device registration](#lwm2m-bulk-device-registration)

For both, navigate to **Devices** > **Registration** in the Device Management application and click **Register device** at the top right.

![LWM2M device registration dropdown](/images/device-protocols/lwm2m/lwm2m-device-reg-dropdown.png)

<a name="lwm2m-single-device-registration"></a>
### Single device registration

> To register a LWM2M device in {{< product-c8y-iot >}} use the LWM2M device registration form by navigating to **Devices** > **Registration** in the Device Management application and click **Register device** > **LWM2M** at the top right.

Reviewer 3: ```suggestion
To register a LWM2M device in {{< product-c8y-iot >}} navigate to **Devices** > **Registration** in the Device Management application, click **Register device** at the top right and select **LWM2M** from the dropdown.
```

The resulting device registration wizard has three steps.

> The first covers is mandatory settings for establishing a connection.

Reviewer 3: ```suggestion
The first covers mandatory settings for establishing a connection.
```

Refer to [Mandatory settings](#lwm2m-device-registration-mandatory-settings) below for details about the fields.

> ![LWM2M single device registration form - mandatory](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page1.png)

Reviewer 3: I strongly recommend you to remove all screenshots which only show the registration wizard that I already see on my screen, or any success or error messages. We need to maintain all these images and update them as soon as there is a design change in the UI (color, for example). If the images do not show specific configurations they are not of much value to the user.

If PSK security mode is selected, additional settings are required.

![LWM2M single device registration form - mandatory with psk](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page1-psk.png)

The second step covers the configuration of the bootstrap settings.
This configuration enables the device to connect to the Cumulocity LWM2M bootstrap server.
Refer to [Bootstrap settings](#lwm2m-device-registration-bootstrap-settings) below for details about the fields.

![LWM2M single device registration form - bootstrap](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page2-1.png)
![LWM2M single device registration form - bootstrap 2](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page2-2.png)

The third step covers optional configurations.
Refer to [Advanced settings](#lwm2m-device-registration-advanced-settings) below for details about the fields.

> ![LWM2M single device registration form - advanced 1](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-1.png)

Reviewer 3: See comment above

![LWM2M single device registration form - advanced 2](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-2.png)
![LWM2M single device registration form - advanced 3](/images/device-protocols/lwm2m/lwm2m-device-reg-single-page3-3.png)

After filling all applicable information for the device, click **Register**.
The UI will display a confirmation message.

> ![LWM2M single device registration form - success](/images/device-protocols/lwm2m/lwm2m-device-reg-single-success.png)

Reviewer 3: See comment above. Why is this screenshots important?

<a name="lwm2m-bulk-device-registration"></a>
### Bulk device registration

If there is a number of devices to be registered at the same time, it is more convenient to use bulk device registration.

> To register the LWM2M devices in {{< product-c8y-iot >}}, navigate to **Devices** > **Registration** in the Device Management application and click **Register device** > **Bulk device registration** > **LWM2M** at the top right.

Reviewer 3: ```suggestion
To register the LWM2M devices in {{< product-c8y-iot >}}, navigate to **Devices** > **Registration** in the Device Management application, click **Register device** at the top right, and select **Bulk device registration** > **LWM2M** from the dropdown.
```

### Device registration settings

<a name="lwm2m-device-registration-mandatory-settings"></a>
#### Mandatory settings

The fields below must be contained to be able to establish a connection:
<table>
<col style="width:25%">
<col style="width:10%">
<col style="width:50%">
<col style="width:15%">
<thead>
<tr>
<th style="text-align:left">Field / CSV field name</th>
<th style="text-align: left">Type</th>
<th style="text-align:left">Description</th>
<th style="text-align: left">Mandatory</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">Endpoint client ID / endpoint id</td>
<td style="text-align: left">String</td>
<td style="text-align:left">Indicates the LWM2M client's endpoint ID in order to allow the LWM2M bootstrap to provision the bootstrap information for the LWM2M client. The endpoint ID has be to be <b>unique</b> across all tenants and must have the same value as the ID. Registering a device using an endpoint ID already used will result in an error.</td>
<td style="text-align: left">Mandatory</td>
</tr>
<tr>
<td style="text-align:left">Security mode / securityMode</td>
<td style="text-align: left">String</td>

> <td style="text-align:left">Determines the type of connection used by the LWM2M device. "NO_SEC" is used for unsecure connections which means that there is no security. It is highly recommended to always protect the LWM2M protocol. However, there are scenarios in which the LWM2M protocol is deployed in environments where the lower layer security mechanisms are provided. "PSK" is used for secure connections. If PSK is chosen, devices need to connect to the LWM2M server using DTLS with the given pre-shared key (PSK). With "PSK", the client and server have a common secret symmetric cryptography. Currently {{< product-c8y-iot >}} supports only "NO_SEC" and "PSK".</td>

Reviewer 3: ```suggestion
<td style="text-align:left">Determines the type of connection used by the LWM2M device. "NO_SEC" is used for unsecure connections which means that there is no security. It is highly recommended to always protect the LWM2M protocol. However, there are scenarios in which the LWM2M protocol is deployed in environments where the lower layer security mechanisms are provided. "PSK" is used for secure connections. If PSK is selected, devices need to connect to the LWM2M server using DTLS with the given pre-shared key (PSK). With "PSK", the client and server have a common secret symmetric cryptography. Currently {{< product-c8y-iot >}} supports only "NO_SEC" and "PSK".</td>
```

<col style="width:15%">
<thead>
<tr>
<th style="text-align: left">Field / CSV field name</th>
<th style="text-align: left">Type</th>
<th style="text-align: left">Description</th>
<th style="text-align: left">Mandatory</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left">LWM2M server URI / lwm2m server uri</td>
<td style="text-align: left">String</td>

> <td style="text-align: left">The URI the server is using for bootstrap. The LWM2M bootstrap server is used to provision the LWM2M client with the information required to contact the LWM2M servers. If you are using the {{< product-c8y-iot >}} service, the hostname of the LWM2M server is "lwm2m.{{< domain-c8y >}}". The bootstrap ports are "5683" for unsecure bootstrap connections and "5684" for secure bootstrap connections. The LWM2M server ports are "5783" for unsecure server connections and "5784" for secure server connections. Note that these values can be different for other services.</td>

Reviewer 3: ```suggestion
<td style="text-align: left">The URI the server uses for bootstrap. The LWM2M bootstrap server is used to provision the LWM2M client with the information required to contact the LWM2M servers. If you use the {{< product-c8y-iot >}} service, the hostname of the LWM2M server is "lwm2m.{{< domain-c8y >}}". The bootstrap ports are "5683" for unsecure bootstrap connections and "5684" for secure bootstrap connections. The LWM2M server ports are "5783" for unsecure server connections and "5784" for secure server connections. Note that these values can be different for other services.</td>
```

<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left">LWM2M bootstrap short server ID / bootstrapShortServerId</td>
<td style="text-align: left">Integer</td>
<td style="text-align: left">The short server ID to be used for the bootstrap server. Default is "0".</td>
<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left; height: 13px;">Registration lifetime / registrationLifetime</td>
<td style="text-align: left">Integer</td>
<td style="text-align: left">The registration lifetime that is sent to the device during bootstrap. Overrides global agent configuration. The value must be specified in seconds.</td>
<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left">Default minimum period / defaultMinimumPeriod</td>
<td style="text-align: left">Integer</td>

> <td style="text-align: left">The default minimum period to configure during bootstrap. See the LWM2M Spec for explanation.</td>

Reviewer: What's the LWM2M Spec?
can we link to it? (multiple occurrences)

<td style="text-align: left">Boolean</td>
<td style="text-align: left">See the <a href="http://www.openmobilealliance.org/release/lightweightm2m/V1_0-20170208-A/OMA-TS-LightweightM2M-V1_0-20170208-A.pdf">LWM2M specification</a>. Allowed values are true or false. Default: Not configured.</td>
<td style="text-align: left">Optional, defaults to Leshan default behavior</td>
</tr>
<tr>
<td style="text-align: left">LWM2M request timeout / lwm2mRequestTimeout</td>
<td style="text-align: left">Integer</td>
<td style="text-align: left">The timeout used for shell operation requests such as read, write, execute done by the LWM2M microservice to the LWM2M device.
The value is in milliseconds and can be given to override the default value that is provided in the LWM2M microservice property file with the "C8Y.lwm2mRequestTimeout" property.
The value must not exceed the maximum request timeout limit given in the LWM2M microservice property file with the "C8Y.lwm2mMaxRequestTimeout" property.
<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left">Binary delivery encoding / binaryDeliveryEncoding</td>
<td style="text-align: left">String</td>

> <td style="text-align: left">Indicates the encoding format for writing binaries to a LWM2M device. The encoding format can be "OPAQUE", "TLV", "JSON" or "TEXT". In case of empty or invalid entries, the default format is "OPAQUE".

Reviewer 3: ```suggestion
<td style="text-align: left">Indicates the encoding format for writing binaries to a LWM2M device. The encoding format can be OPAQUE, TLV, JSON or TEXT. In case of empty or invalid entries, the default format is OPAQUE.
```

<td style="text-align: left">Optional, defaults to false</td>
</tr>
</tbody>
</table>

The following table explains several optional parameters related to firmware updates which help in tuning the Firmware Over The Air (FOTA) parameters on a device level.

<table>
<col style="width:25%">
<col style="width:10%">
<col style="width:50%">
<col style="width:15%">
<thead>
<tr>
<th style="text-align: left">Field / CSV field name</th>
<th style="text-align: left">Type</th>
<th style="text-align: left">Description</th>
<th style="text-align: left">Mandatory</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left">Firmware update delivery method / fwUpdateDeliveryMethod</td>
<td style="text-align: left">String</td>
<td style="text-align: left">Explains the firmware update delivery method.

> Allowed values are "PUSH", "PULL" or "BOTH".</td>

Reviewer 3: ```suggestion
Allowed values are PUSH, PULL or BOTH.</td>
```

<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left">Firmware update supported device protocol / fwUpdateSupportedDeviceProtocol</td>
<td style="text-align: left">String</td>
<td style="text-align: left">Indicates the device protocol to be used for the firmware update. Allowed values are "COAP", "COAPS", "HTTP" or "HTTPS".</td>
<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left">Firmware update reset mechanism / fwUpdateResetMechanism</td>
<td style="text-align: left">String</td>
<td style="text-align: left">Indicates the mechanism used to reset the firmware update state machine.

> Allowed values are "PACKAGE" or "PACKAGE_URI". Depending on the value, the LWM2M agent either writes an empty string to package URI resource or sets the package resource to NULL ('\0'). If this field is not used, the default reset state machine mechanism is used where a reset is done via package resource for PUSH and via package URI for PULL.</td>

Reviewer 3: ```suggestion
Allowed values are PACKAGE or PACKAGE_URI. Depending on the value, the LWM2M agent either writes an empty string to package URI resource or sets the package resource to NULL ('\0'). If this field is not used, the default reset state machine mechanism is used where a reset is done via package resource for PUSH and via package URI for PULL.</td>
```

If the value is not set, the awake time is determined by the LWM2M client's registration awake time attribute "at" or, if this attribute is also not found, then by the global setting that is defined in the LWM2M microservice.
</td>
<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left; height: 40px;">Notification storing when disabled or offline / notificationIfDisabled</td>
<td style="text-align: left">Boolean</td>
<td style="text-align: left">See the <a href="http://www.openmobilealliance.org/release/lightweightm2m/V1_0-20170208-A/OMA-TS-LightweightM2M-V1_0-20170208-A.pdf">LWM2M specification</a>. Allowed values are true or false. Default: Not configured.</td>
<td style="text-align: left">Optional, defaults to Leshan default behavior</td>
</tr>
<tr style="height: 40px;">
<td style="text-align: left">Disable timeout / disableTimeout</td>
<td style="text-align: left">Boolean</td>
<td style="text-align: left">See the <a href="http://www.openmobilealliance.org/release/lightweightm2m/V1_0-20170208-A/OMA-TS-LightweightM2M-V1_0-20170208-A.pdf">LWM2M specification</a>. Allowed values are true or false. Default: Not configured.</td>

> <td style="text-align: left">Optional, defaults to Leshan default behavior</td>

Reviewer 3: ```suggestion
<td style="text-align: left">Optional, defaults to Leshan default behavior.</td>
```

<td style="text-align: left">Indicates the encoding format for writing binaries to a LWM2M device. The encoding format can be "OPAQUE", "TLV", "JSON" or "TEXT". In case of empty or invalid entries, the default format is "OPAQUE".
<td style="text-align: left">Optional</td>
</tr>
<tr>
<td style="text-align: left">Use common timestamp resources 5518 and 6050 or object specific timestamp for object 6 reported by the device if available / enableResourceLevelTimestamp</td>
<td style="text-align: left">Boolean</td>

> <td style="text-align: left">When this device property is enabled, then {{< product-c8y-iot >}} is registering the device's reported data either the time send from the device in the common timestamp resources 5518 and 6050 or the specific timestamp resource for object 6. The timestamp is used only per Object ID reported in a single or composite device response. Using a timestamp from other objects is not possible. Otherwise {{< product-c8y-iot >}} will use its local date and time to register the resource data. Allowed values are true or false. Default: false.

Reviewer 4: This is a little confusing, because the sentences are rather long I'd phrase it like this:

If this device property is enabled {{< product-c8y-iot >}} uses time stamps reported by the device for constructing measurements, events and alarms.  This is possible if the device reports multiple resources at the same time. For example, this may be done using observes on entire object instances or composite reads and observes. Time stamp data needs to be reported as part of the same object: Either via the common timestamp resources 5518 and 6050 or the specific timestamp resource for object 6. If this feature is disabled or no timestamp information is reported, {{< product-c8y-iot >}} will use its local date and time. Default: false
